signer.js

```diff
@@ -42,6 +72,42 @@ class SignatureSigner {
   }
 }
 
+/**
+ * This class is used for signing certificates using elliptic library
+ */
+class EllipticSignatureSigner extends SignatureSigner {
+  /**
+   * @param {string} pem
+   */
+  constructor(pem) {
+    super(pem);
+  }
+
+  /**
+   * This method is used for signing the content
+   * @param {Uint8Array} content
+   * @return {Uint8Array}
+   */
+  sign(content) {
+    try {
+      const base64 = this.getBase64PemString();
+      const {prvKeyHex} = jsrsasign.KEYUTIL.getKey(
+          jsrsasign.b64utohex(base64), null, 'pkcs5prv');
+      const EC = elliptic.ec;
+      const ecdsaCurve = elliptic.curves['p256'];
+      const ecdsa = new EC(ecdsaCurve);
+      const signKey = ecdsa.keyFromPrivate(prvKeyHex, 'hex');
+      const sha256 = crypto.createHash('sha256');
+      const digest = sha256.update(content).digest();
+      const signature = ecdsa.sign(Buffer.from(digest, 'hex'), signKey);
+      return new Uint8Array(signature.toDER());
+    } catch (err) {
+      throw new Error(`Failed to sign the request ${err}`);
+    }
+  }
+}
+
 module.exports = {
-  SignatureSigner,
+  SignatureSigner: EllipticSignatureSigner, // Use elliptic library as default
+  JsrsasignSignatureSigner,
```

Comment:
elliptic is using deterministic k value while jsrsasign seems to be using non-deterministic one. As a result, elliptic creates same signature from same input data. That's the difference with jsrsasign. I think it's usual for blockchains as other blockchains such as Bitcoin and Hyperledger fabric is also using deterministic way. But it might be better to be able to choose either one for users.

@feeblefakie We have 3 options. Do you think which one is the best?
1. Remove JsrsasignSignatureSigner completely
2. Comment out JsrsasignSignatureSigner and keep it for future
3. Add a selector based on the properties in constructor ClientServiceBase (e.g., scalar.ledger.client.signer = 'jsrsasign')

Reply:
I also noticed elliptic always generate the same signatures for the same contents.
It doesn't really matter unless we consider reply attacks.
However, since the communication is protected by TLS,
considering replay attacks might be too excessive.

I am not sure whether Scalar DL server uses contracts or consensus-like protocol to validate signatures. By far, I only test with the case in which there is only 1 Scalar DL server.
If it is, I agree with you. Using a deterministic algorithm is better.